Add localisation for future translation (En + Fr)

aelisya opened the pull request:
I have added the language management and translated the extension in French (the extension will automatically change language depending on the active localization in the browser (the default one is English in case the said language is not available in the translations already done), 

I also retrieved the version number from the manifest so you won't have to manually change the version number in popup.html in the next releases.
The text of popup.html is added via javascript in popup.js
@mlgualtieri

Commit:
Localisation

Localisation of the manifest

Commit:
English + French

Add local for english and french

Commit:
Add js for using localisation

Replace hard coded text with local one for future translation of the addon

Commit:
Change popup for remove hardcoded text

remove hard coded text and add id for insert localised one.

Commit:
Update manifest.json

same modification than the chrome one

Commit:
Update popup.html

Same modification than the chrome one

Commit:
Update popup.js

Same modification than the chrome one

Commit:
Add files via upload

same local from the chrome one

mlgualtieri commented:
Awesome, thanks!  I'll need to take your word that the translations are accurate :-)

I'll merge this into the next release, which will hopefully be later this week.  Lot's of bugfixes and improvements in the forthcoming 1.0.18 release.

aelisya commented:
for french yes it's accurate, i can do other languages, but if i do it will be with deepl translator (so i prefer let a native speaker create these languages to be sure it accurate).

mlgualtieri commented:
I agree on that point.  Let's just keep it with English/French for now, and if there's demand for other languages maybe users can supply translation.

aelisya commented:
at least after the merging, the translation will be very easy, and if you add a new thing on the popup all you have to do, you open an issue, and mention me (and other translator later) and we will translate the new thing shortly after ;)

Commit:
Merge branch 'master' into master

aelisya commented:
@mlgualtieri i have updated the merge for take account of your change (but it must be tested), but normally it must work

Commit:
Update manifest.json

corect mistake

Commit:
CSP patch

i recomand using default-src 'self' because default-src serves as fallback for directives that are absent in CSP (so technically more secure it lock all to 'self').
source : https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src

Commit:
CSP patch

i recomand using default-src 'self' because default-src serves as fallback for directives that are absent in CSP (so technically more secure it lock all to 'self').
source : https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src

aelisya commented:
@mlgualtieri i have added " default-src 'self' " in csp like recommended in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src for directives that are absent in CSP.
When you will update i will check the pull request again (just in case)

mlgualtieri commented:
The CSP patch sounds fine.  I'll need to test to make sure it doesn't have an ill effects, but doesn't seem like it would.  I need to hold off merging in the new language edits until all the new settings are in place, since it will add new text to popup.html.  I'm expecting to release what's currently in github today, ahead of Chrome 85, and will have some other updates soon in the next release that will add more settings.

aelisya commented:
Focus on your thing, i will correct my pull request when you are done ^^
So later you can do a .1 or .2 with translation.

Commit:
Merge branch 'master' into master

ghost commented:
Conflict resolved (will test if all work well after a little sleep xD)

edit: some translate to do will do that too after little sleep ^^

Commit:
Update popup.js

Adding the traduction injection for 1.1

Commit:
Update popup.html

Adding some id for injection of translation

Commit:
Update messages.json

adding missing translation for english 1.1

Commit:
Update messages.json

adding mission translation for french 1.1

Commit:
Update popup.js

adding missing injection for 1.1 message

Commit:
Update popup.html

adding some id to translation for 1.1

Commit:
Update messages.json

updated for 1.1 english

Commit:
Update messages.json

update for 1.1 French

Commit:
Update popup.html

update forgotten suppresion of the version (not necessary anymore with the translation mod who retreive the version from manifest.json

Commit:
Update manifest.json

forget to remove a line

ghost commented:
@mlgualtieri patch + translation done to 1.1.0 update , tested on firefox and vivaldi (chromium), but since my browser is in french i don't have tested english version (but if it work in french it must work in english)

Commit:
Update manifest.json

adding forgotten connect-src

Commit:
Update manifest.json

adding forgotten connect-src

ghost commented:
patch to the CSP for add connect-src https://* to it now it rework fine

Commit:
Update manifest.json

correct mistake

Commit:
Update manifest.json

add data: to autorise base64

Commit:
Update manifest.json

add data: to autorise base64

ghost commented:
to correct csp error when encounter a base 64 picture i have added data: to csp under connect-src to manifest json

Commit:
Update manifest.json

transfer data to img-src in the csp (error of data created on the option page from extension)

Commit:
Update manifest.json

transfer data to img-src in the csp (error of data created on the option page from extension)

ghost commented:
transfer data: to img-src in the csp (error of data: created on the option page from extension)

ghost commented:
@mlgualtieri just to report since the last push on manifest i don't have trigger any error in the console (Vivaldi), if someone else want to try, before an eventual merge it will be nice (since i don't really use problematic website (the website who usually trigger an issue in this repository), i think more testing must be do to be sure my push don't break anything (especially the CSP in manifest.json)

ghost commented:
@mlgualtieri any news on this ?

mlgualtieri commented:
I haven't had a chance to work on the plugin since the last release.  Although I've received no errors with the current CSP in manifest.json, so it might be a better idea to leave that edit unchanged when I roll in this patch.

Commit:
Update manifest.json

Commit:
Update manifest.json

ghost commented:
@mlgualtieri Sorry for the delay, the csp is reset at the actual version in your repository like you have asked, you can test but normally it work.
